File: julia/loops.jl
import Dates

function simulate()
  start = Dates.now()
  iterations = 500_000
  sample_size = 23

  count = 0
  for i in 1:iterations
    data = rand(1:365, 1, sample_size)
    if length(unique(data)) != sample_size
      count += 1
    end
  end
  println("iterations: ", iterations)
  println("sample-size: ", sample_size)
  percent = count / iterations * 100
  println("percent: ", round(percent,digits=2))
  finish = Dates.now()
  diff = (finish - start) |> Dates.value |> x->x/1000
  println("seconds: ", diff)
end

simulate()

File: php/loops.php
<?php
  function simulate() {
    $start = microtime(true);
    $iterations = 1000000;
    $sample_size = 23;

    $count = 0;
    for ($x = 0; $x < $iterations; $x++) {
      $data = array_fill(0, 365, 0);
      for ($n = 0; $n < $sample_size; $n++) {
        $number = rand(0, 364);
        if ($data[$number] === 1) {
          $count++;
          break;
        } else {
          $data[$number] = 1;
        }
      }
    }
    print "iterations: $iterations\n";
    print "sample-size: $sample_size\n";
    $percent = round($count / $iterations * 100, 2);
    print "percent: $percent\n";
    $end = microtime(true);
    $seconds = round($end - $start, 3);
    print "seconds: $seconds\n";
  }

  simulate();
?>
